components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -2332,6 +2335,29 @@ public static JWT encryptJWT(JWTClaimsSet jwtClaimsSet, JWEAlgorithm encryptionA
         }
     }
 
+    /**
+     * This is the generic Encryption function which calls algorithm specific encryption function
+     * depending on the algorithm name.
+     *
+     * @param signedJwt           contains signed JWT body
+     * @param encryptionAlgorithm JWT encryption algorithm
+     * @param spTenantDomain      Service provider tenant domain
+     * @param clientId            ID of the client
+     * @return encrypted JWT token
+     * @throws IdentityOAuth2Exception
+     */
+    public static JWT encryptJWT(String signedJwt, JWEAlgorithm encryptionAlgorithm,
```

Comment:
Any particular reason the signed JWT was passed as a string here?

Reply:
Any particular reason why we opted expected the signing to happen outside of this method and not within this method?

Reply:
fixed in 4d887a5